components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java

```diff
@@ -755,4 +758,43 @@ public static boolean isRecaptchaEnabledForConnector(IdentityGovernanceService i
         }
         return Boolean.parseBoolean(enable);
     }
+
+    /**
+     * Check whether generic ReCaptcha validation is enabled for the given authenticator.
+     * This method is used by the jsp pages to check whether the ReCaptcha validation is enabled for the current
+     * authenticator.
+     *
+     * @param authenticatorName Name of the authenticator.
+     * @return True if generic ReCaptcha validation is enabled.
+     */
+    public static boolean isGenericRecaptchaEnabledAuthenticator(String authenticatorName) {
+
+        if (StringUtils.isBlank(authenticatorName)) {
+            return false;
+        }
+        AuthenticatorConfig authenticatorConfig =
+                ConfigurationFacade.getInstance().getAuthenticatorConfig(authenticatorName);
+
+        Map<String, String> params = authenticatorConfig.getParameterMap();
+
+        return params != null && params.get(ENABLE_GENERIC_CAPTCHA_VALIDATION) != null &&
+                Boolean.parseBoolean(params.get(ENABLE_GENERIC_CAPTCHA_VALIDATION));
```

Comment:
Simplify the logic, because `params.get(ENABLE_GENERIC_CAPTCHA_VALIDATION) != null` is redundant.